ext/lb/haproxy/haproxy.go

```diff
@@ -43,7 +44,17 @@ func (p *HAProxyLoadBalancer) ConfigPath() string {
 }
 
 func (p *HAProxyLoadBalancer) Template() string {
-	return haproxyConfTemplate
+	d, err := ioutil.ReadFile(p.cfg.TemplatePath)
+
+	if err == nil {
+		return string(d)
+	} else {
+		log().Infof("Missing haproxy configuration template: file=%s", p.cfg.TemplatePath)
```

Comment:
Good idea. This got me thinking it doesn't fail fast enough too. 

I moved the check for this up to `lb.NewLoadBalancer` so it's DRY and fails faster.